iroh-sync/src/keys.rs

```diff
@@ -0,0 +1,276 @@
+//! Keys used in iroh-sync
+
+use std::{cmp::Ordering, fmt, str::FromStr};
+
+use ed25519_dalek::{Signature, SignatureError, Signer, SigningKey, VerifyingKey};
+use rand_core::CryptoRngCore;
+use serde::{Deserialize, Serialize};
+
+/// Author key to insert entries in a [`crate::Replica`]
+///
+/// Internally, an author is a [`SigningKey`] which is used to sign entries.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct Author {
+    signing_key: SigningKey,
+}
+impl Author {
+    /// Create a new [`Author`] with a random key.
+    pub fn new<R: CryptoRngCore + ?Sized>(rng: &mut R) -> Self {
+        let signing_key = SigningKey::generate(rng);
+        Author { signing_key }
+    }
+
+    /// Create an [`Author`] from a byte array.
+    pub fn from_bytes(bytes: &[u8; 32]) -> Self {
+        SigningKey::from_bytes(bytes).into()
+    }
+
+    /// Returns the [`Author`] byte representation.
+    pub fn to_bytes(&self) -> [u8; 32] {
+        self.signing_key.to_bytes()
+    }
+
+    /// Returns the [`AuthorId`] byte representation.
+    pub fn id_bytes(&self) -> [u8; 32] {
+        self.signing_key.verifying_key().to_bytes()
+    }
```

Comment:
Seems redundant since `AuthorId::as_bytes` already exists

Reply:
Yup I agree, @dignifiedquire added it, are you fine with removing it again?

Reply:
Removed it